code/game/objects/structures/flora.dm

```diff
@@ -78,6 +78,15 @@
 	icon = 'icons/obj/plants.dmi'
 	icon_state = "plant-26"
 
+//Added random icon selection for potted plants.
+//It was silly they always used the same sprite when we have 26 sprites of them in the icon file
+/obj/structure/flora/pottedplant/New()
+	..()
+	var/number = rand(1,26)

```

Comment:
The same can be said of every single thing in the game with an icon. If you change the names of itemstates or dmi files, yes things will break. Always, in every case.

If someone is going to change a dmi, its their responsibility to also update anything that references it